storage/storage_sqlite.go

```diff
@@ -260,6 +263,46 @@ func (s *SQL) Purge(oldest time.Time) (uint64, error) {
 }
 
 /*
+GetBackup creates a backup of the current database in a temp filename,
+and write the persistent db file to the specified writer.
+*/
+func (s *SQL) GetBackup(w io.Writer) (err error) {
+	backupDbName := fmt.Sprintf("./transicator_sqlite_backup_%v_%v", rand.Int(), time.Now().UnixNano())
+	log.Debugf("Sqlite backup destination: %s", backupDbName)
+
+	// create backup
+	bc := s.Backup(backupDbName)
+	for {
+		br := <-bc
+		log.Debugf("Backup %d remaining\n", br.PagesRemaining)
+		if err = br.Error; err != nil {
+			log.Error(err)
+			return
+		}
+		if br.Done {
+			break
+		}
+	}
+
+	// remove the temp backup file
+	defer os.RemoveAll(backupDbName)
```

Comment:
Why not defer this before calling "s.Backup"? Otherwise, what happens if the backup fails part-way -- we will leave the file there and never remove it.